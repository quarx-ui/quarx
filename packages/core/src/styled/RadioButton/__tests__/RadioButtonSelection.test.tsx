import { cleanup, render, screen } from '@testing-library/react';
import React from 'react';
import userEvent from '@testing-library/user-event';
import { RadioButtonSelection } from '@core';

describe('RadioButtonSelection', () => {
    it('default snapshot', () => {
        const { asFragment } = render(<RadioButtonSelection title="RadioButtonSelection" />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('text should be in the document', () => {
        render(<RadioButtonSelection title="RadioButtonSelection" />);

        expect(screen.queryByText('RadioButtonSelection')).toBeInTheDocument();
        const radioButton = document.querySelector('input');
        expect(radioButton).not.toHaveAttribute('disabled');
    });

    it('should be checked', () => {
        render(<RadioButtonSelection checked title="RadioButtonSelection" />);

        const radioButton = document.querySelector('input');
        expect(radioButton?.checked).toEqual(true);
    });

    it('should have attribute disabled', () => {
        const mockedOnClick = jest.fn();
        render(<RadioButtonSelection disabled onClick={mockedOnClick} title="RadioButtonSelection" />);
        const input = document.querySelector('input');
        const radioButton = document.querySelector('.QxRadioButtonSelection');

        if (!radioButton || !input) { return; }

        expect(input).toHaveAttribute('disabled');
        userEvent.click(radioButton);
        expect(mockedOnClick).not.toBeCalled();
    });

    it('should be hidden', () => {
        render(<RadioButtonSelection hidden title="RadioButtonSelection" />);
        expect(document.querySelector('.QxRadioButtonSelection')).not.toBeInTheDocument();

        cleanup();

        render(<RadioButtonSelection permissions={{ hidden: true }} title="RadioButtonSelection" />);
        expect(document.querySelector('.QxRadioButtonSelection')).not.toBeInTheDocument();
    });

    it('onClick should be called', () => {
        let checked = false;
        const mockedOnChange = jest.fn(() => {
            checked = !checked;
        });
        render(<RadioButtonSelection onClick={mockedOnChange} title="RadioButtonSelection" />);
        const radioButton = document.querySelector('.QxRadioButtonSelection');

        if (!radioButton) { return; }

        userEvent.click(radioButton);
        expect(mockedOnChange).toBeCalledTimes(1);
        expect(checked).toEqual(true);
    });
});
