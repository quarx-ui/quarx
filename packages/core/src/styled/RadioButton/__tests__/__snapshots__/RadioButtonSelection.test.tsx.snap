// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtonSelection default snapshot 1`] = `
<DocumentFragment>
  <div
    class="QxSelection QxSelection_size_medium QxSelection_styleHover_default QxSelection_borderRadius_smooth QxRadioButtonSelection css-1e56tu3-QxSelection-root"
  >
    <div
      class="QxSelection-leftAdornment css-1xv7td9-QxSelection-leftAdornment"
    >
      <label
        class="QxRadioButton QxRadioButton_size_medium QxRadioButton_color_brand QxRadioButton_disableFocus QxRadioButton_position_left css-144omum-QxRadioButton-root"
      >
        <input
          class="QxRadioButton-input css-l1mh22-QxRadioButton-input"
          type="radio"
          value=""
        />
        <div
          class="QxRadioButton-marker css-n013xa-QxRadioButton-marker"
        >
          <div
            class="QxRadioButton-markerDot css-1cuw3c2-QxRadioButton-markerDot"
          />
        </div>
      </label>
    </div>
    <div
      class="QxSelection-text css-4p6ohz-QxSelection-text"
    >
      <div
        class="QxSelection-title css-11y2um3-QxSelection-title"
      >
        RadioButtonSelection
      </div>
    </div>
  </div>
</DocumentFragment>
`;
